diff --git a/server/application/application.go b/server/application/application.go
index d3ada7707..6988e9bb4 100644
--- a/server/application/application.go
+++ b/server/application/application.go
@@ -1163,7 +1163,7 @@ func (s *Server) getAppProject(ctx context.Context, a *v1alpha1.Application, log
 func (s *Server) Delete(ctx context.Context, q *application.ApplicationDeleteRequest) (*application.ApplicationResponse, error) {
 	appName := q.GetName()
 	appNs := s.appNamespaceOrDefault(q.GetAppNamespace())
-	a, _, err := s.getApplicationEnforceRBACClient(ctx, rbac.ActionGet, q.GetProject(), appNs, appName, "")
+	a, _, err := s.getApplicationEnforceRBACClient(ctx, rbac.ActionDelete, q.GetProject(), appNs, appName, "")
 	if err != nil {
 		return nil, err
 	}
@@ -1171,10 +1171,6 @@ func (s *Server) Delete(ctx context.Context, q *application.ApplicationDeleteReq
 	s.projectLock.RLock(a.Spec.Project)
 	defer s.projectLock.RUnlock(a.Spec.Project)
 
-	if err := s.enf.EnforceErr(ctx.Value("claims"), rbac.ResourceApplications, rbac.ActionDelete, a.RBACName(s.ns)); err != nil {
-		return nil, err
-	}
-
 	if q.Cascade != nil && !*q.Cascade && q.GetPropagationPolicy() != "" {
 		return nil, status.Error(codes.InvalidArgument, "cannot set propagation policy when cascading is disabled")
 	}
@@ -2068,7 +2064,7 @@ func isTheSelectedOne(currentNode *v1alpha1.ResourceNode, q *application.Applica
 
 // Sync syncs an application to its target state
 func (s *Server) Sync(ctx context.Context, syncReq *application.ApplicationSyncRequest) (*v1alpha1.Application, error) {
-	a, proj, err := s.getApplicationEnforceRBACClient(ctx, rbac.ActionGet, syncReq.GetProject(), syncReq.GetAppNamespace(), syncReq.GetName(), "")
+	a, proj, err := s.getApplicationEnforceRBACClient(ctx, rbac.ActionSync, syncReq.GetProject(), syncReq.GetAppNamespace(), syncReq.GetName(), "")
 	if err != nil {
 		return nil, err
 	}
@@ -2083,10 +2079,6 @@ func (s *Server) Sync(ctx context.Context, syncReq *application.ApplicationSyncR
 		return a, status.Errorf(codes.PermissionDenied, "cannot sync: blocked by sync window")
 	}
 
-	if err := s.enf.EnforceErr(ctx.Value("claims"), rbac.ResourceApplications, rbac.ActionSync, a.RBACName(s.ns)); err != nil {
-		return nil, err
-	}
-
 	if syncReq.Manifests != nil {
 		if err := s.enf.EnforceErr(ctx.Value("claims"), rbac.ResourceApplications, rbac.ActionOverride, a.RBACName(s.ns)); err != nil {
 			return nil, err
diff --git a/server/application/application_test.go b/server/application/application_test.go
index ae9de1d65..bce8b1037 100644
--- a/server/application/application_test.go
+++ b/server/application/application_test.go
@@ -836,6 +836,8 @@ func TestNoAppEnumeration(t *testing.T) {
 		require.EqualError(t, err, common.PermissionDeniedAPIError.Error(), "error message must be _only_ the permission error, to avoid leaking information about app existence")
 		_, err = appServer.Get(adminCtx, &application.ApplicationQuery{Name: new("doest-not-exist"), Project: []string{"test"}})
 		assert.EqualError(t, err, "rpc error: code = NotFound desc = applications.argoproj.io \"doest-not-exist\" not found", "when the request specifies a project, we can return the standard k8s error message")
+		_, err = appServer.Get(adminCtx, &application.ApplicationQuery{Name: new("test"), Project: []string{"wrong-project"}})
+		assert.EqualError(t, err, "rpc error: code = NotFound desc = applications.argoproj.io \"test\" not found", "when the request specifies a project and it mismatches, we return the standard k8s error message")
 	})
 
 	t.Run("GetManifests", func(t *testing.T) {
@@ -984,6 +986,25 @@ func TestNoAppEnumeration(t *testing.T) {
 		require.EqualError(t, err, common.PermissionDeniedAPIError.Error(), "error message must be _only_ the permission error, to avoid leaking information about app existence")
 		_, err = appServer.Sync(adminCtx, &application.ApplicationSyncRequest{Name: new("doest-not-exist"), Project: new("test")})
 		assert.EqualError(t, err, "rpc error: code = NotFound desc = applications.argoproj.io \"doest-not-exist\" not found", "when the request specifies a project, we can return the standard k8s error message")
+		_, err = appServer.Sync(adminCtx, &application.ApplicationSyncRequest{Name: new("test"), Project: new("wrong-project")})
+		assert.EqualError(t, err, "rpc error: code = NotFound desc = applications.argoproj.io \"test\" not found", "when the request specifies a project and it mismatches, we return the standard k8s error message")
+	})
+
+	t.Run("Delete", func(t *testing.T) {
+		_, _ = appServer.Create(adminCtx, &application.ApplicationCreateRequest{Application: &v1alpha1.Application{ObjectMeta: metav1.ObjectMeta{Name: "test-delete"}, Spec: v1alpha1.ApplicationSpec{Project: "default"}}})
+		
+		// Use cascade=false to avoid finalizer patch in tests
+		falseVar := false
+		_, err := appServer.Delete(adminCtx, &application.ApplicationDeleteRequest{Name: new("test-delete"), Cascade: &falseVar})
+		require.NoError(t, err)
+		_, err = appServer.Delete(noRoleCtx, &application.ApplicationDeleteRequest{Name: new("test"), Cascade: &falseVar})
+		require.EqualError(t, err, common.PermissionDeniedAPIError.Error(), "error message must be _only_ the permission error, to avoid leaking information about app existence")
+		_, err = appServer.Delete(adminCtx, &application.ApplicationDeleteRequest{Name: new("doest-not-exist"), Cascade: &falseVar})
+		require.EqualError(t, err, common.PermissionDeniedAPIError.Error(), "error message must be _only_ the permission error, to avoid leaking information about app existence")
+		_, err = appServer.Delete(adminCtx, &application.ApplicationDeleteRequest{Name: new("doest-not-exist"), Project: new("test"), Cascade: &falseVar})
+		assert.EqualError(t, err, "rpc error: code = NotFound desc = applications.argoproj.io \"doest-not-exist\" not found", "when the request specifies a project, we can return the standard k8s error message")
+		_, err = appServer.Delete(adminCtx, &application.ApplicationDeleteRequest{Name: new("test"), Project: new("wrong-project"), Cascade: &falseVar})
+		assert.EqualError(t, err, "rpc error: code = NotFound desc = applications.argoproj.io \"test\" not found", "when the request specifies a project and it mismatches, we return the standard k8s error message")
 	})
 
 	t.Run("TerminateOperation", func(t *testing.T) {
